source/extensions/filters/http/dynamic_modules/filter_config.h

```diff
@@ -74,6 +78,46 @@ class DynamicModuleHttpFilterConfig {
   OnHttpFilterScheduled on_http_filter_scheduled_ = nullptr;
 
   Envoy::Upstream::ClusterManager& cluster_manager_;
+  const Stats::ScopeSharedPtr stats_scope_;
+  Stats::StatNamePool stat_name_pool_;
+  const Stats::StatName custom_stat_namespace_;
+  // We only allow the module to create stats during envoy_dynamic_module_on_http_filter_config_new,
+  // and not later during request handling, so that we don't have to wrap the stat storage in a
+  // lock.
+  bool stat_creation_frozen_ = false;
+
+  size_t addCounter(Stats::Counter& counter) {
+    size_t id = counters_.size();
+    counters_.push_back(counter);
+    return id;
+  }
+
+  Stats::Counter& getCounterById(size_t id) const {
+    ASSERT(id < counters_.size());
```

Comment:
note: ASSERT is only available on the debug build so doing not so much as users(module owners) are likely to work with release builds, but yeah harmless.